### PR TITLE
feat: Add account authority transfer wrapper

### PR DESCRIPTION
**Pull request by exdx**
Adds the ability to create the new transfer account authority ix from the TS API.

See https://github.com/mrgnlabs/marginfi-v2/pull/163 for more information on the implementation.

Signed-off-by: denton <denton24646@gmail.com>

### CODE REVIEW

File: packages/marginfi-client-v2/src/models/account/pure.ts [hunk 1]
```diff
@@ -942,6 +947,20 @@ class MarginfiAccount {
...
 
+  async makeAccountAuthorityTransferIx(program: MarginfiProgram, newAccountAuthority: PublicKey) : Promise<InstructionsWrapper> {
+    const accountAuthorityTransferIx = await instructions.makeAccountAuthorityTransferIx(
+      program,
+      {
+        marginfiAccountPk: this.address,
+        marginfiGroupPk: this.group,
+        signerPk: this.authority,
+        newAccountAuthorityPk: newAccountAuthority,
+        feePayerPk: this.address, // ?
```
Comment (losman0s) [inline packages/marginfi-client-v2/src/models/account/pure.ts:958]:
```suggestion
        feePayerPk: this.authority,
```

In the context of the SDK, let's simply set this to that for now. This won't be compatible with squads but we can fix that at that time, as I don't know off-hand the logic needed when this gets used through the MS.

**Review by losman0s — DISMISSED** [review]

File: packages/marginfi-client-v2/src/models/account/pure.ts (hunk 1 above)
Comment (exdx) [reply packages/marginfi-client-v2/src/models/account/pure.ts:958]: But do we want the fee payer to be the same as the signer? I imagine a scenario where we could pay for the account authority transfer on behalf of the user?

**Comment by exdx** [comment]
The IDL is currently broken -- it needs to be auto-generated and added in the correct instruction

File: packages/marginfi-client-v2/src/idl/marginfi.json [hunk 2]
```diff
@@ -1306,7 +1533,7 @@
           {
             "name": "accountFlags",
             "docs": [
-              "The flas that indicates the state of the account.",
+              "The flags that indicate the state of the account.",
```
[the remark this answers is not in the document]
Comment (losman0s) [reply packages/marginfi-client-v2/src/idl/marginfi.json:1536]: docstring? yes

File: packages/marginfi-client-v2/src/idl/marginfi.json [hunk 3]
```diff
@@ -1891,10 +2118,8 @@
             }
           },
           {
-            "name": "oracleKeys",
-            "type": {
-              "array": ["publicKey", 5]
-            }
+            "name": "oracleKey",
+            "type": "publicKey"
```
[the remark this answers is not in the document]
Comment (losman0s) [reply packages/marginfi-client-v2/src/idl/marginfi.json:2122]: Technically this is a breaking change, but one localized to an admin instruction, so only drawback would be to have to
- adapt any client making use of it (e.g. the squads UI which is anyway not ready on that side), and
- use a versioned approach to indexing past txs using this struct, which is a very minor use case

So it's fine

**Pull request merged**